### PR TITLE
GKS hackathon changes tracking

### PR DESCRIPTION
**Pull request by theferrit32**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +29,12 @@ environment.
     make devready
     source venv/3.12/bin/activate
 
+
+Set up the `pre-commit` hook
+
+    cp ./scripts/pre-commit ./.git/hooks/
```
Comment (korikuzma) [inline README.md:35]: Is there a reason why we're not leveraging pre-commit directly? https://pre-commit.com/

File: pyproject.toml [hunk 2]
```diff
@@ -53,3 +60,28 @@ source2classes = "ga4gh.gks.metaschema.scripts.source2classes:cli"
...
 requires = ["setuptools>=65.3", "setuptools_scm>=8"]
 build-backend = "setuptools.build_meta"
+
+
+[tool.ruff]
+line-length = 120
+target-version = "py311"
+
+[tool.ruff.lint]
+# select = ["E4", "E7", "E9", "F"]
```
Comment (korikuzma) [inline pyproject.toml:70]:
```suggestion 
```

File: pyproject.toml [hunk 3]
```diff
@@ -53,3 +60,28 @@ source2classes = "ga4gh.gks.metaschema.scripts.source2classes:cli"
 [build-system]
 requires = ["setuptools>=65.3", "setuptools_scm>=8"]
 build-backend = "setuptools.build_meta"
+
+
+[tool.ruff]
+line-length = 120
+target-version = "py311"
```
Comment (korikuzma) [inline pyproject.toml:67]: We require python >= 3.12
```suggestion
target-version = "py312"
```

File: src/templates/maturity
Comment (korikuzma) [inline src/templates/maturity:1]: nice

File: README.md (hunk 1 above)
Comment (theferrit32) [reply README.md:35]: For simple stuff like just running some ruff commands, adding another tool dependency seems like a lot. But I'd be fine with someone else adding it if they want

**Pull request merged**